CenterForOpenScience/osf.io
# [ENG-1976] Meetings Autoban

## Purpose

Ensure akismet is not overzealous with it's bans of meetings users for spam.

## Changes
Functional:
   - add conditional to check if a user is a meetings user or not during `check_spam` method
   - add auto_check_spam field for meetings and add migrations
   - add tests

Non-functional
   - mocked responses in a better less global way
   - allows admin app to change spam check.

## QA Notes

Low risk, small migration. Expected behavior is that a person can create a meetings account and join normally without being marked as spam and banned, if auto_check_spam is set to false.

Also minor new admin app behavior with auto_spam_check toggle pictured here:
<img width="1204" alt="Screen Shot 2020-07-06 at 10 17 51 AM" src="https://user-images.githubusercontent.com/9688518/86604671-ae061580-bf73-11ea-9037-3504a4c73f4e.png">

## Documentation

🐞 fix, no docs.

## Side Effects

None that I know of.

## Ticket

https://openscience.atlassian.net/browse/ENG-1976
